### PR TITLE
Enhancements and code simplifications in various files

### PR DESCRIPTION
**Pull request by timosachsenberg**
## Description

Continues https://github.com/OpenMS/OpenMS/pull/7308

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Enhanced the False Discovery Rate (FDR) calculation by adding new options for score selection and orientation, and improved protein FDR calculation with group filtering.
- Introduced new methods in `IDScoreSwitcherAlgorithm` for retrieving and switching score types, along with improved documentation.
- Simplified code in `MSstatsFile` by removing unused variables and using more efficient methods.
- Added options for peptide and protein score filtering in `IDFilter`, integrating `IDScoreSwitcherAlgorithm` for score type validation.
- Updated score type setting for peptide identifications in `SageAdapter`.
- Corrected typos and improved documentation across several files.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Miscellaneous</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>FalseDiscoveryRate.cpp</strong><dd><code>Remove debug logging for parameter settings</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/ANALYSIS/ID/FalseDiscoveryRate.cpp

- Removed debug code related to parameter logging.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-dce3903183b4c89431e937e86f54f38e0a5a25391dea6b357150f76fe81f479e">+0/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>7 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>IDScoreSwitcherAlgorithm.cpp</strong><dd><code>Add method to retrieve score type names</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/ANALYSIS/ID/IDScoreSwitcherAlgorithm.cpp

- Added a method to retrieve score type names.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-8e46c4350794bd369157a16f86d606d9235ba69ae70193c363af1ab2b1a16d0d">+15/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MSstatsFile.cpp</strong><dd><code>Simplify code by removing unused variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/FORMAT/MSstatsFile.cpp

<li>Simplified code by removing unused variable.<br> <li> Replaced tuple creation with emplace method.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-84c58d52d8f0b0f1d9e495d1237c14b81d12927ba9606e3a54a18a90ad989172">+1/-11</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>FalseDiscoveryRate.cpp</strong><dd><code>Enhance FDR calculation with new score options</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/topp/FalseDiscoveryRate.cpp

<li>Added new options for score selection and orientation.<br> <li> Enhanced protein FDR calculation with group filtering.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-033f4c544133fb5af32546517a305d5a89ec8a6656727dd829b3108ca5a5df17">+53/-5</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IDFilter.cpp</strong><dd><code>Add score filtering options for peptides and proteins</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/topp/IDFilter.cpp

<li>Added options for peptide and protein score filtering.<br> <li> Integrated IDScoreSwitcherAlgorithm for score type validation.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-1f9d296d610e31782ad1a26609f859451dfc735448ee2c4c11c0b6108c61f243">+45/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>SageAdapter.cpp</strong><dd><code>Update score type setting for peptide identifications</code>&nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/topp/SageAdapter.cpp

- Updated score type setting for peptide identifications.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-e7a6640c1869e5ae5c09daa8f8cf1545eb7f920b62d8e829bbf5cf9aadff1b92">+3/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IDScoreSwitcherAlgorithm.h</strong><dd><code>Enhance score switching with new methods and documentation</code></dd></summary>
<hr>

src/openms/include/OpenMS/ANALYSIS/ID/IDScoreSwitcherAlgorithm.h

<li>Added methods for score type retrieval and switching.<br> <li> Enhanced documentation for score switching methods.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-ab9f484fd25ca0ab4aa22b655b7e5e89dae910f86a1ed96d1deca200913e751b">+149/-16</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>IDFilter.h</strong><dd><code>Add predicates for meta value filtering and enhance documentation</code></dd></summary>
<hr>

src/openms/include/OpenMS/PROCESSING/ID/IDFilter.h

<li>Added new predicates for meta value filtering.<br> <li> Enhanced documentation for filtering predicates.<br>

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-551b8cf5af907a3a1534ef80b20ed5ca34e1ce4c87cf864e218d6188d895eef5">+129/-4</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>IDFilter.cpp</strong><dd><code>Correct typo in comment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

src/openms/source/PROCESSING/ID/IDFilter.cpp

- Corrected a typo in a comment.

</details>

  </td>
  <td><a href="https://github.com/OpenMS/OpenMS/pull/7642/files#diff-f8ec592ff77b68da57bbbda129f5e57e9f7bf1de21e8a511dcf9076e129879ee">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: src/topp/FalseDiscoveryRate.cpp [hunk 1]
```diff
@@ -89,8 +89,21 @@ class TOPPFalseDiscoveryRate :
     setValidStrings_("PSM", ListUtils::create<String>("true,false"));
     registerStringOption_("peptide", "<FDR level>", "false", "Perform FDR calculation on peptide level and annotates it as meta value\n(Note: if set, also calculates FDR/q-value on PSM level.)", false);
     setValidStrings_("peptide", ListUtils::create<String>("true,false"));
+    registerStringOption_("PSM_peptide_base_score", "<score name or type>", "", "Set if you want to choose a different score than the last calculated main score for PSM or peptide level.", false);
```
Comment (timosachsenberg) [inline src/topp/FalseDiscoveryRate.cpp:92]: @jpfeufer might need some additional documentation. why PSM and peptide? etc.

File: src/topp/FalseDiscoveryRate.cpp [hunk 2]
```diff
@@ -89,8 +89,21 @@ class TOPPFalseDiscoveryRate :
...
+    registerStringOption_("PSM_peptide_base_score_orientation", "<higher/lower>", "", "In case the score orientation cannot be inferred.", false, true);
+    setValidStrings_("PSM_peptide_base_score_orientation", ListUtils::create<String>("higher_better, lower_better"));
     registerStringOption_("protein", "<FDR level>", "true", "Perform FDR calculation on protein level", false);
     setValidStrings_("protein", ListUtils::create<String>("true,false"));
+    registerStringOption_("protein_group", "<FDR level>", "false", "Perform FDR calculation on (indist.) protein group level, too. Currently, this will enable protein FDR automatically (since internals need to be in-sync) but will affect the level at which it filters (if enabled).", false);
```
Comment (timosachsenberg) [inline src/topp/FalseDiscoveryRate.cpp]: rename to proteingroup to be consistent with the one in IDFilter?

File: src/topp/IDFilter.cpp [hunk 3]
```diff
@@ -127,9 +128,18 @@ class TOPPIDFilter :
...
+    registerStringOption_("score:type_peptide", "<type>", "", "Score used for filtering. If empty, the main score is used.", false, true);
+    setValidStrings_("score:type_peptide", ids.getScoreTypeNames());
+
     registerDoubleOption_("score:protein", "<score>", NAN, "The score which should be reached by a protein hit to be kept. All proteins are filtered based on their singleton scores irrespective of grouping. Use in combination with 'delete_unreferenced_peptide_hits' to remove affected peptides. (use 'NAN' to disable this filter)", false);
+    registerStringOption_("score:type_protein", "<type>", "", "The type of the score which should be reached by a protein hit to be kept. If empty, the most recently set score is used.", false, true);
+    setValidStrings_("score:type_protein", ids.getScoreTypeNames());
+
     registerDoubleOption_("score:proteingroup", "<score>", NAN, "The score which should be reached by a protein group to be kept. Performs group level score filtering (including groups of single proteins). Use in combination with 'delete_unreferenced_peptide_hits' to remove affected peptides. (use 'NAN' to disable this filter)", false);
```
Comment (timosachsenberg) [inline src/topp/IDFilter.cpp:141]: see here

File: src/topp/FalseDiscoveryRate.cpp (hunk 1 above)
Comment (jpfeuffer) [reply src/topp/FalseDiscoveryRate.cpp:92]: Well it obviously depends on the parameter right before it. Not sure if that needs to made any clearer.

File: src/topp/FalseDiscoveryRate.cpp (hunk 2 above)
Comment (jpfeuffer) [reply src/topp/FalseDiscoveryRate.cpp]: Fine with me

**Pull request merged**